problem_builder/public/js/mentoring_with_steps.js

```diff
@@ -217,11 +251,20 @@ function MentoringWithStepsBlock(runtime, element) {
         reviewButtonDOM.hide();
         tryAgainDOM.show();
 
+        /**
+         * We detach review step from DOM, this is required to handle HTML
+         * blocks that can be added to the Review step.
+         *
+         * NOTE: This is handled differently than step js. As the html contents
+         * of review step are replaced with fresh contents in submit function.  
```

Comment:
@jbzdak "Review steps are handled differently than "normal" steps: The HTML contents of a review step are replaced with ..."